framework/operator.go

```diff
@@ -0,0 +1,205 @@
+package framework
+
+import (
+	"github.com/cenk/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+// Config represents the configuration used to create a new operator framework.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new operator
+// framework by best effort.
+func DefaultConfig() Config {
+	var err error
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	return Config{
+		// Dependencies.
+		Logger: newLogger,
+	}
+}
+
+// New creates a new configured operator framework.
+func New(config Config) (*Framework, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	newFramework := &Framework{
+		// Dependencies.
+		logger: config.Logger,
+	}
+
+	return newFramework, nil
+}
+
+type Framework struct {
+	// Dependencies.
+	logger micrologger.Logger
+}
+
+// ProcessCreate is a drop-in for an informer's AddFunc. It receives the custom
+// object observed during TPR watches and anything that implements Resource.
+// ProcessCreate takes care about all necessary reconciliation logic for create
+// events.
+//
+//     func addFunc(obj interface{}) {
+//         err := f.ProcessCreate(obj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         AddFunc:    addFunc,
+//     }
+//
+func (f *Framework) ProcessCreate(obj interface{}, resources []Resource) error {
```

Comment:
The resources have been variadic before. I changed this because of the interface changes when providing a backoff.